core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -396,8 +407,11 @@ public void insertRecord(
       logger.info("Spilling data because number of spilledRecords crossed the threshold " +
         numElementsForSpillThreshold);
       spill();
+    } else if (inMemRecordsSize >= maxRecordsSizeForSpillThreshold) {
+      logger.info("Spilling data because size of spilledRecords crossed the threshold " +
```

Comment:
Should we also include the number of records and threshold here?

Reply:
+1, let's follow https://github.com/apache/spark/pull/24618/files#diff-3eedc75de4787b842477138d8cc7f150R413